src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -1345,6 +1345,135 @@ class BSE(ExcitedStateMethodology):
+        """,
+    )
+
+    is_tamm_dancoff = Quantity(
+        type=bool,
+        description='If True, Tamm-Dancoff approximation (TDA) is used (linear-response only).',
+    )
+
+    solver = Quantity(
+        type=MEnum(
+            'Casida', 'Sternheimer', 'Liouville-Lanczos', 'propagation', 'unavailable'
+        ),
+        description="""
+        Numerical driver:
+          • Casida             : eigenvalue problem in particle-hole space
+          • Sternheimer        : frequency-domain Sternheimer
+          • Liouville-Lanczos  : iterative Liouville-Lanczos
+          • propagation        : explicit time evolution (real-time)
+          • unavailable        : code-chosen / unspecified
+        """,
+    )
+
+    xc_kernel_ref = Quantity(
+        type=XCFunctional,
+        description="""
+        Reference to an `XCFunctional` describing the TDDFT kernel.
+        Typically points to the ground-state DFT XC section if the adiabatic
+        kernel matches, or to a separate XCFunctional created for the TD kernel
+        when it differs (e.g., LRC/hybrid kernel on top of semi-local ground state).
+        """,
+    )
+
+    external_field_ref = Quantity(
+        type=Photon,
+        description='External field / polarization used to drive the response or propagation.',
+    )
+
+    target_property = Quantity(
+        type=MEnum('absorption', 'emission', 'EELS', 'Raman', 'nonlinear'),
+        description='Intended spectral/response target of the TDDFT input.',
+    )
+
+    time_step = Quantity(
+        type=np.float64,
+        unit='second',
+        description='Time step used for real-time propagation.',
+    )
+
+    n_steps = Quantity(
+        type=np.int32,
+        description='Number of time steps in real-time propagation.',
+    )
+
+    propagator = Quantity(
+        type=MEnum(
+            'Crank-Nicolson',
+            'ETRS',
+            'Magnus',
+            'Runge-Kutta',
+            'Lanczos',
+            'split_operator',
+        ),
+        description='Time integrator used for real-time propagation.',
+    )
+
+    gauge = Quantity(
+        type=MEnum('length', 'velocity'),
+        description='Electromagnetic gauge used for coupling to the field (mostly real-time).',
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+        self.name = 'TDDFT'
+
+        # Soft warnings to help parsers spot missing essentials
+        if self.type == 'real_time' and (
+            self.time_step is None or self.n_steps is None
+        ):
+            logger.warning(
+                'TDDFT real_time mode without time_step or n_steps may be incomplete.'
+            )
+        if self.type == 'linear_response' and self.is_tamm_dancoff is None:
+            logger.warning(
+                'TDDFT linear_response set but is_tamm_dancoff not specified.'
+            )
```

Comment:
You aren't covering the other case, where `is_tamm_dancoff` is set, but `type == real-time`.
If the edge cases become sufficiently complex, I would in fact add a test. You can then move this logic out to a dedicated function.